accepted/00000-saltclass-external-pillar.md

```diff
@@ -0,0 +1,147 @@
+- Feature Name: Saltclass as external pillar mechanism
+- Start Date: 2019-08-20
+- RFC PR:
+
+# Summary
+[summary]: #summary
+
+Enhance/replace current custom external pillar mechanism with upstream provided Saltclass module.
+
+# Motivation
+[motivation]: #motivation
+
+SUSE Manager needs a way how to export data about systems, groups, services etc. to the salt configuration management. In current implementation this is done by custom python script [suma_minion.py](https://github.com/uyuni-project/uyuni/blob/master/susemanager-utils/susemanager-sls/modules/pillar/suma_minion.py) used by salt master leveraging its external pillar mechanism. This worked so far very well, even allowing Uyuni some extra tricks like integration of Formulas with Forms mechanism. However the implementation has also some limitations preventing further growth.
+
+This RFC proposes to, at first, enable yet another external pillar mechanism in the salt, but this time not a custom script, but salt's own [saltclass](https://docs.saltstack.com/en/latest/ref/pillar/all/salt.pillar.saltclass.html) mechanism.
+
+Examples of limitations of our external pillar implementation:
+- allows only static pillars (no jinja preparsing, no pillar dependencies on other pillars)
+- problematic data merging
+- made integration of formulas easy, but causes duplication of code where i.e. form.yml is parsed on both java and python level
+
+# Detailed design
+[design]: #detailed-design
+
+There are three choices how to address the introduction of saltclass to the SUSE Manager:
+
+A) Enable concurrent usage of both custom `suma_minion.py` and `saltclass` external pillars
+B) Enable concurrent usage of both custom `suma_minion.py` and `saltclass` external pillars, but with the target to gradually migrate from `suma_minion.py` to `saltclass`.
+C) Do the one time migration and switch from custom `suma_minion.py` to `saltclass`
+
+## What means migrating from suma_minion.py to saltclass?
+
+Even in its relative simplicity `suma_minion.py` does more then just loading `yaml` files and exporting to the `salt` ecosystem. `suma_minion.py` is also integral part of the Formulas with Forms enablement in the SUSE Manager/Uyuni.
+
+`suma_minion.py` responsibilities regarding formulas are:
+- read what formulas are enabled for given minion and prepare `formulas` grain which is then used by `mgr_tops`, respectively `formulas.sls` loader to apply formulas as part of highstate
+- read formulas `json` data and `form.yml` and construct complete pillar information
+
+Complete migration from `suma_minoni.py` to `saltclass` then means:
+- implement Formulas with Forms enablement in other way then in external pillar, preferably in java code itself as the current implementation causes triplication of the formulas enablement code
```

Comment:
I don't quite get the comment about "triplication" of the enablement code. Look at how it's done with custom/custom_groups/custom_org, the code used for Salt-based configuration channels. It isn't any more elegant.
But yes, those two need to be unified as they're basically doing the same thing (adding states to a highstate).